src/runtime/test/json-id-to-manifest-test.ts

```diff
@@ -0,0 +1,131 @@
+/**
+ * @license
+ * Copyright (c) 2019 Google Inc. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * Code distributed by Google as part of this project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import {assert} from '../../platform/chai-web.js';
+import {JsonldToManifest, supportedTypes} from '../converters/jsonldToManifest.js';
+import {Predicate} from '../hot.js';
+import {Manifest} from '../manifest.js';
+import fs from 'fs';
+
+describe('JsonldToManifest', () => {
+
+  const isValidManifest = (manifestStr: string): boolean => {
+    try {
+      Manifest.parse(manifestStr);
+    } catch (error) {
+      return false;
+    }
+    return true;
+  };
+
+  const getSchema = async (schema: string = 'Product'): Promise<string> => {
+    return new Promise<string>((resolve, reject) => {
```

Comment:
Can use node's promisify if you want Promise based readfile.

Reply:
2b6d2bd